src/kili/client.py

```diff
@@ -39,6 +40,17 @@
 warnings.filterwarnings("default", module="kili", category=DeprecationWarning)
 
 
+class FilterPoolFullWarning(logging.Filter):
+    """Filter out the specific urllib3 warning related to the connection pool."""
+
+    def filter(self, record) -> bool:
+        """urllib3.connectionpool:Connection pool is full, discarding connection: ..."""
+        return "Connection pool is full, discarding connection" not in record.getMessage()
+
+
+logging.getLogger("urllib3.connectionpool").addFilter(FilterPoolFullWarning())
+
+
```

Comment:
and creating a class like this which is not called anywhere is enough for filtering ?

Reply:
it worked for me, can  you try it too? Since you had the same issue apparently

Reply:
works for me too